superset/models/helpers.py

```diff
@@ -1341,36 +1342,34 @@ def get_time_filter(  # pylint: disable=too-many-arguments
         return and_(*l)
 
     def values_for_column(self, column_name: str, limit: int = 10000) -> list[Any]:
-        """Runs query against sqla to retrieve some
-        sample values for the given column.
-        """
-        cols = {}
-        for col in self.columns:
-            if isinstance(col, dict):
-                cols[col.get("column_name")] = col
-            else:
-                cols[col.column_name] = col
-
-        target_col = cols[column_name]
-        tp = None  # todo(hughhhh): add back self.get_template_processor()
+        # always denormalize column name before querying for values
+        db_dialect = self.database.get_dialect()
+        denomalized_col_name = self.database.db_engine_spec.denormalize_name(
```

Comment:
denomarlized name will return the original value if it can't be adjusted by the engine:
[preset-io/superset@60e1526/superset/db_engine_specs/base.py#L1950](https://github.com/preset-io/superset/blob/60e1526f6acc5cbfd126a29ac7d6af9420f8a648/superset/db_engine_specs/base.py#L1950)